Objects/structseq.c

```diff
@@ -409,16 +395,51 @@ PyStructSequence_InitType(PyTypeObject *type, PyStructSequence_Desc *desc)
 PyTypeObject*
 PyStructSequence_NewType(PyStructSequence_Desc *desc)
 {
-    PyTypeObject *result;
+    PyObject* type;
+    PyObject* bases;
+    PyMemberDef* members;
+
+    PyType_Spec* spec = PyMem_NEW(PyType_Spec, 1);
+    spec->name = desc->name;
```

Comment:
Is this safe?  I think it is okay because PyType_FromSpecWithBases() will allocate a new PyUnicode object, copying the string data and then set ht_name.

Reply:
Yes, this is totally safe. It turns out that PyType_FromSpecWithBases has most of its use cases of the spec name to be read-only (such as error messages during the execution of the function). The only instance where it requires to persist it, it creates a PyUnicode object.